From 6b6167c33a2053a94a664ec27a07a3f77c854757 Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Mon, 27 Nov 2023 07:50:58 +0200
Subject: [PATCH] [FSW-334] version: rename preboot defines

As preboot app has been removed, version components are renamed
to ppboot

Change-Id: Ia9a7d966fe852517c03f9d0d7644ff77cff8f4d2
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/Makefile b/Makefile
index 29ef5bc..b5cc2e7 100644
--- a/Makefile
+++ b/Makefile
@@ -77,12 +77,12 @@
 	@$(CC) $(LDFLAGS) -T $(PPBOOT_LD_SCRIPT) -o $(BINDIR)/$@ -Wl,--start-group $(PPBOOT_OBJECTS) $(LDLIBS) -Wl,--end-group
 
 $(AUTO_GENERATED_FILE): localversion-habanalabs
-	@( printf '#define PREBOOT_VER "%s"\n' '$(shell cut -d- -f2- localversion-habanalabs)' ) > $@
-	@( printf '#define PREBOOT_MAJOR_VER %d\n' '$(shell cut -d'-' -f4 localversion-habanalabs | cut -d'.' -f1)' ) >> $@
-	@( printf '#define PREBOOT_MINOR_VER %d\n' '$(shell cut -d'-' -f4 localversion-habanalabs | cut -d'.' -f2)' ) >> $@
-	@( printf '#define PREBOOT_PATCH_VER %d\n' '$(shell cut -d'-' -f4 localversion-habanalabs | cut -d'.' -f3)' ) >> $@
-	@LANG=en_us date +'#define PREBOOT_DATE "%b %d %Y"' >> $@
-	@date +'#define PREBOOT_TIME "%T"' >> $@
+	@( printf '#define PPBOOT_VER "%s"\n' '$(shell cut -d- -f2- localversion-habanalabs)' ) > $@
+	@( printf '#define PPBOOT_MAJOR_VER %d\n' '$(shell cut -d'-' -f4 localversion-habanalabs | cut -d'.' -f1)' ) >> $@
+	@( printf '#define PPBOOT_MINOR_VER %d\n' '$(shell cut -d'-' -f4 localversion-habanalabs | cut -d'.' -f2)' ) >> $@
+	@( printf '#define PPBOOT_PATCH_VER %d\n' '$(shell cut -d'-' -f4 localversion-habanalabs | cut -d'.' -f3)' ) >> $@
+	@LANG=en_us date +'#define PPBOOT_DATE "%b %d %Y"' >> $@
+	@date +'#define PPBOOT_TIME "%T"' >> $@
 
 buildversion-habanalabs: include/baremetal.h $(AUTO_GENERATED_FILE)
 	echo "VERSION_STR" | gcc -include $< -I./ -I ./lib/common/ -I ./$(HL_DEV_ARCH)/ -I$(SPECS_EMBEDDED_DIR)/$(SPECS_EMB_DEV) -I$(SPECS_DIR)/$(BASE_DEV_NAME)/asic_reg -I$(SPECS_DIR)/ -I ./include/ -P -E -o - - | tail -1 | sed 's/\" \+\"//g;s/["\n]//g' | cut -d' ' -f2- | tr -d '\n' > $@
diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 3fcc759..1780113 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -1288,8 +1288,8 @@
 
 void ppboot_send_version(void)
 {
-	if (ppboot_boot_info_set(HL_BOOT_INFO_PPBOOT_VERSION, (void *)PREBOOT_VER,
-				 sizeof(PREBOOT_VER)))
+	if (ppboot_boot_info_set(HL_BOOT_INFO_PPBOOT_VERSION, (void *)PPBOOT_VER,
+				 sizeof(PPBOOT_VER)))
 		preboot_puts("BOOT_INFO: Failed to set ppboot version\n");
 }
 
diff --git a/include/baremetal.h b/include/baremetal.h
index 741ee11..f8d75b9 100644
--- a/include/baremetal.h
+++ b/include/baremetal.h
@@ -58,8 +58,8 @@
 
 #define VERSION_MAX_LEN		128
 
-#define VERSION_STR	"version " PREBOOT_VER \
-	" (" PREBOOT_DATE " - " PREBOOT_TIME ")"
+#define VERSION_STR	"version " PPBOOT_VER \
+	" (" PPBOOT_DATE " - " PPBOOT_TIME ")"
 
 #ifndef __packed
 #define __packed	__attribute__((packed))
diff --git a/lib/common/module_i2c_slave.c b/lib/common/module_i2c_slave.c
index cbda9cf..156d9b6 100644
--- a/lib/common/module_i2c_slave.c
+++ b/lib/common/module_i2c_slave.c
@@ -288,13 +288,13 @@
 		ret = HL_VERSIONS_CMD_LEN;
 		break;
 	case HL_VERSIONS_FW_MAJOR_VER_VAL:
-		ret = PREBOOT_MAJOR_VER;
+		ret = PPBOOT_MAJOR_VER;
 		break;
 	case HL_VERSIONS_FW_MINOR_VER_VAL:
-		ret = PREBOOT_MINOR_VER;
+		ret = PPBOOT_MINOR_VER;
 		break;
 	case HL_VERSIONS_FW_ADDITIONAL_VER_VAL:
-		ret = PREBOOT_PATCH_VER;
+		ret = PPBOOT_PATCH_VER;
 		break;
 	default:
 		ret = 0;
